src/gleam/crypto.gleam

```diff
@@ -39,6 +39,23 @@ pub type HashAlgorithm {
 @external(javascript, "../gleam_crypto_ffi.mjs", "hash")
 pub fn hash(a: HashAlgorithm, b: BitArray) -> BitArray
 
+pub type HashState
+
+/// Starts hash algorithm for streaming
+@external(erlang, "gleam_crypto_ffi", "hash_init")
+@external(javascript, "../gleam_crypto_ffi.mjs", "hashInit")
+pub fn hash_init(a: HashAlgorithm) -> HashState
+
+/// Starts hash algorithm for streaming
+@external(erlang, "crypto", "hash_update")
+@external(javascript, "../gleam_crypto_ffi.mjs", "hashUpdate")
+pub fn hash_update(state: HashState, data: BitArray) -> HashState
+
+/// Starts hash algorithm for streaming
+@external(erlang, "crypto", "hash_final")
+@external(javascript, "../gleam_crypto_ffi.mjs", "hashFinal")
+pub fn hash_final(state: HashState) -> BitArray
```

Comment:
Not sure about these names. How about `Hasher`, `hasher`, `hash_chunk`, and `digest`?

Reply:
Done